Updated the converter

YifanC opened the pull request:
1. Reversed "traj_id" to the TrackId from edepsim (See S10 here: https://docs.dunescience.org/cgi-bin/private/RetrieveFile?docid=29612&filename=2x2_analysis_meeting_2x2_simulation_calibration_16nov2023.pdf&version=1)
2. Corrected the "pdg_id" in segments dataset (See Issue 33)
3. Store all primary tracks even if they have no descendants in the 2x2 detector
4. Added a new attribute "primary" to label the primary particles.

For 3. 
<img width="579" alt="Screenshot 2023-12-14 at 2 42 38 PM" src="https://github.com/DUNE/2x2_sim/assets/25907864/5e4682a6-eefb-40ab-903b-597ce45247b4">

Commit:
change traj_id to the edepsim TrackId

Commit:
correct the segments' pdg_id

Commit:
make sure all the primary particles are stored

Commit:
add primary particle flag

mjkramer commented on run-convert2h5/convert_edepsim_roottoh5.py:
```diff
@@ -405,28 +405,29 @@ def dump(input_file, output_file, keep_all_dets=False):
+                        # Trace back in the family tree
                         while trajectory.GetParentId() >= -1:
-                            if trackMap[trajectory.GetTrackId()] in trajectories["traj_id"]:
+                            if trajectory.GetTrackId() in trajectories["traj_id"]:
```
Should this be:

```python
if trackMap[trajectory.GetTrackId()] in trajectories["file_traj_id"]:
```

since `traj_id` is no longer unique in `trajectories`?

Or if we want to avoid using `file_traj_id`, then shouldn't we check for an element of `trajectories` that matches both the `vertex_id` and the `traj_id`?

mjkramer replied:
nevermind, `traj_id` is unique in `trajectories` because `trajectories` is reinitialized for every edepsim event (i.e. vertex)

mjkramer commented on run-convert2h5/convert_edepsim_roottoh5.py:
```diff
@@ -405,28 +405,29 @@ def dump(input_file, output_file, keep_all_dets=False):
-                    seg_local_traj_id = hitSegment.Contrib[0]
+                    segment[iHit]["traj_id"] = hitSegment.Contrib[0]
+                    seg_traj_id = hitSegment.Contrib[0]
                     if segment[iHit]["traj_id"] not in trajectories["traj_id"]:
```
See comment below

mjkramer committed:
Add `file_traj_id` to `segments`

To minimize disruption to existing code that relies on having a
file-unique trajectory ID in `segments`. Should eventually deprecate
in order to support merged files.
